jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -476,6 +476,9 @@ JibContainer containerize(
       Containerizer containerizer, Supplier<ExecutorService> defaultExecutorServiceFactory)
       throws IOException, CacheDirectoryCreationException, InterruptedException, RegistryException,
           ExecutionException {
+    if (containerizer.isOfflineMode() && containerizer.getImageConfiguration().isOnlineImage()) {
```

Comment:
I see this is the only place where you want to check `isOnlineImage()` (essentially if it was `Containerizer.to(RegistryImage)`). Since the check is within `Containerizer`, I think we don't have to add `isOnlineImage` to `ImageConfiguration`. Instead, `Containerizer` can have an additional field like `mustBeOnline`, initialized to true only when `RegistryImage`, and you can fail early in `setOfflineMode()`. Do you think this will work?